### PR TITLE
Input

### PR DESCRIPTION
**Pull request by kenshen112**
This is the start of Controller support added to DobieStation. This currently supports Xinput and evdev.

Current Limitations, Steam won't respond to xinput so if you don't have an xbox controller you'll have to use DS4 Windows, evdev lacks a configurator so you'll have to hard code buttons.

For Linux:
look for a package called evtest and install it. running it will tell you the controllers supported button codes as well as the codes for the sticks and triggers. Open linux_input hpp and cpp and map the buttons accordingly.

### CODE REVIEW

**Comment by RedDevilus** [comment]
https://github.com/Ryochan7/DS4Windows/releases

This is a fork for DS4Windows with loads of fixes and additions. Its what i currently use.

File: DobieStation/Core.vcxproj.filters [hunk 1]
```diff
@@ -421,5 +421,7 @@
     <None Include="$(RootDir)LICENSE" />
     <None Include="$(RootDir).gitignore" />
     <None Include="$(RootDir).github\*" />
+    <None Include="$(RootDir).github\*" />
+    <None Include="$(RootDir).github\*" />
```
Comment (Margen67) [inline DobieStation/Core.vcxproj.filters]: ???

File: DobieStation/Core.vcxproj [hunk 2]
```diff
@@ -230,10 +215,8 @@
...
   </ItemGroup>
-
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
-
   <ImportGroup Label="ExtensionTargets">
     <Import Project="$(VCTargetsPath)\BuildCustomizations\masm.targets" />
   </ImportGroup>
-</Project>
\ No newline at end of file
+</Project>
```
Comment (Margen67) [inline DobieStation/Core.vcxproj]: This didn't need to be committed.

File: DobieStation/DobieQt.vcxproj [hunk 3]
```diff
@@ -115,6 +99,5 @@
     <ClCompile Include="$(QtMocOutPrefix)settingswindow.cpp" />
     <ClCompile Include="$(QtMocOutPrefix)gamelistwidget.cpp" />
   </ItemGroup>
-
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
 </Project>
```
Comment (Margen67) [inline DobieStation/DobieQt.vcxproj]: Neither did this.

File: DobieStation/InputCommon.vcxproj.filters [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    <None Include="$(RootDir).gitignore" />
+    <None Include="$(RootDir)CONTRIBUTING.md" />
+    <None Include="application\application.pro" />
+  </ItemGroup>
+  <ItemGroup>
+    <ResourceCompile Include="InputCommon.rc">
+      <Filter>Resource Files</Filter>
+    </ResourceCompile>
+  </ItemGroup>
+</Project>
```
Comment (Margen67) [inline DobieStation/InputCommon.vcxproj.filters]: Needs a newline.

**Comment by Margen67** [comment]
Please don't make another PR for this. It's unnecessary.

**Pull request closed without merging**